doc/proposals/named-networks.md

```diff
@@ -0,0 +1,185 @@
+# Support for named networks
+
+# Table of contents
+
+- [Introduction](#introduction)
+  - [Goal](#goal-of-this-proposal)
+- [Design](#design)
+  - [Changes in IPAM Config](#changes-in-ipam-config)
+  - [Changes in Modules](#changes-in-modules)
+- [Summary](#summary)
+- [Discussions and Decisions](#discussions-and-decisions)
+
+<hr>
+
+## Introduction
+
+When whereabouts assigns an IP to a Pod this fact is recorded in a document of kind `IPPool` that has its name derived from the CIDR range in question.
+Should the user configure multiple overlapping ranges, it is possible to configure whereabouts to allow assigning duplicate IPs.
+However, since the storage of the assignments is done in a document that is named like the CIDR range, it is not possible to configure *the same CIDR range* twice and have whereabouts assign from the ranges independently.
+
+This is, for example, useful in multi-tenant situations where more than one group is responsible for selecting CIDR ranges.
+
+### Goal of this Proposal
+
+- Allow configuring the same CIDR range multiple times (e.g. in separate multus-`NetworkAttachmentDefinitions`)
+
+<hr>
+
+## Design
+
+The IPAM configuration format would be updated to include an optional field for the name of the network in question.
+If this field is left empty, whereabouts will behave as it does now, without these changes.
+
+### Changes in IPAM Config
+
+This will lead to change in IPAM config something as follows:
+
+<table>
+<tr>
+<th>Old IPAM Config</th>
+<th>Changes</th>
+<th>New IPAM Config</th>
+</tr>
+<tr>
+<td>
+  
+```json
+{
+      "cniVersion": "0.3.0",
+      "name": "whereaboutsexample",
+      "type": "macvlan",
+      "master": "eth0",
+      "mode": "bridge",
+      "ipam": {
+        "type": "whereabouts",
+        "range": "192.168.2.225/28",
+        "exclude": [
+           "192.168.2.229/30",
+           "192.168.2.236/32"
+        ]
+      }
+}
+```
+  
+</td>
+<td>
+
+```diff
+{
+      "cniVersion": "0.3.0",
+      "name": "whereaboutsexample",
+      "type": "macvlan",
+      "master": "eth0",
+      "mode": "bridge",
+      "ipam": {
+        "type": "whereabouts",
+        "range": "192.168.2.225/28",
++       "network_name": "whereaboutsexample",
```

Comment:
I don't think this is needed. What's wrong with using the `name` attribute defined [above](https://github.com/k8snetworkplumbingwg/whereabouts/blob/545f05525c710a1f7f5b34c47ee6396d223581bc/pkg/types/types.go#L44) ?